src/coreclr/jit/lsraxarch.cpp

```diff
@@ -1900,21 +1905,30 @@ int LinearScan::BuildIntrinsic(GenTree* tree)
             break;
     }
     assert(tree->gtGetOp2IfPresent() == nullptr);
+
+// TODO-XARCH-AVX512 this is overly constraining register available as NI_System_Math_Abs
```

Comment:
Can we have this extracted out in a separate method (along with the comment) instead of having this  check at multiple places?

Reply:
Also, shouldn't this be under `isEvexCompatible` check instead of just using `RBM_LOWSIMD` for AMD64?

Reply:
In this case, its the `NamedIntrinsics` and not the `HWIntrinsics`, e.g., those defined in `namedintrinsiclist.h`.

I suppose I can try and add a similar marking for these but as far as I know there is not a set of flags for these named intrinsics?